### PR TITLE
dynamodb: respond with proper error messages

### PR DESCRIPTION
**Pull request by benwaffle**
Without this, KCLv1 for java throws exceptions.

### CODE REVIEW

**Comment by benwaffle** [comment]
I can't figure out which test is failing.

**Comment by winly19** [comment]
I can't figure out which test is failing

**Comment by winly19** [comment]
continuous-integration/travis-ci/pr — The Travis CI build could not complete due to an error

**Comment by benwaffle** [comment]
Yeah I looked at the travis logs but I can't figure out exactly which test if failing.

File: localstack/services/dynamodb/dynamodb_listener.py [hunk 1]
```diff
@@ -53,12 +53,26 @@ def forward_request(self, method, path, data, headers):
...
             if to_str(data['TableName']) not in table_names:
-                return 404
+                response = Response()
+                response.status_code = 400
+                response._content = json.dumps({
+                    '__type': 'com.amazonaws.dynamodb.v20120810#ResourceNotFoundException',
+                    'message': 'Cannot do operations on a non-existent table'
+                })
+                fix_headers_for_updated_response(response)
+                return response
```
Comment (whummer) [inline localstack/services/dynamodb/dynamodb_listener.py]: I believe we could simply return the following:
```
response = error_response(message='Cannot do operations on a non-existent table', 
                          error_type='ResourceNotFoundException')
fix_headers_for_updated_response(response)
return response
```
(same for lines 68-75)

**Comment by whummer** [comment]
Thanks @benwaffle . Re-running the build seems to have worked (looks like we have a flaky test there). Added a minor comment - would be great if we could address it before merging. thx

**Comment by benwaffle** [comment]
@whummer I updated it to use `error_response()`

**Comment by benwaffle** [comment]
Great! Do you know when a new docker image will be released for `localstack/localstack`?

**Comment by whummer** [comment]
@benwaffle The `latest` tag is pushed automatically by our CI system with the latest changes on `master` branch. And we'll cut over a new tagged release in the next couple of days.

**Pull request merged**